homeassistant/components/pterodactyl/sensor.py

```diff
@@ -0,0 +1,208 @@
+
+from .coordinator import PterodactylConfigEntry, PterodactylCoordinator, PterodactylData
+from .entity import PterodactylEntity
+
+KEY_CPU_UTILIZATION = "cpu_utilization"
+KEY_MEMORY_UTILIZATION = "memory_utilization"
+KEY_DISK_UTILIZATION = "disk_utilization"
+KEY_NETWORK_INBOUND = "network_inbound"
+KEY_NETWORK_OUTBOUND = "network_outbound"
+KEY_UPTIME = "uptime"
+
+# Coordinator is used to centralize the data updates.
+PARALLEL_UPDATES = 0
+
+
+@dataclass(frozen=True, kw_only=True)
+class PterodactylSensorEntityDescription(SensorEntityDescription):
+    """Class describing Pterodactyl sensor entities."""
+
+    value_fn: Callable[[PterodactylData], StateType]
+    attributes_fn: Callable[[PterodactylData], dict[str, Any]] | None
+
+
+def get_extra_state_attributes_cpu(
+    data: PterodactylData,
+) -> dict[str, list[str]]:
+    """Return CPU limit as extra state attribute."""
+    extra_state_attributes: dict[str, Any] = {}
+
+    extra_state_attributes["cpu_limit"] = data.cpu_limit
+
+    return extra_state_attributes
+
+
+def get_extra_state_attributes_memory(
+    data: PterodactylData,
+) -> dict[str, list[str]]:
+    """Return memory usage and limit as extra state attributes."""
+    extra_state_attributes: dict[str, Any] = {}
+
+    extra_state_attributes["memory_usage"] = round(
+        InformationConverter.convert(
+            data.memory_usage, UnitOfInformation.BYTES, UnitOfInformation.MEGABYTES
+        )
+    )
+    extra_state_attributes["memory_limit"] = round(
+        InformationConverter.convert(
+            data.memory_limit, UnitOfInformation.BYTES, UnitOfInformation.MEGABYTES
+        )
+    )
+
+    return extra_state_attributes
+
+
+def get_extra_state_attributes_disk(
+    data: PterodactylData,
+) -> dict[str, list[str]]:
+    """Return disk usage and limit as extra state attributes."""
+    extra_state_attributes: dict[str, Any] = {}
+
+    extra_state_attributes["disk_usage"] = round(
+        InformationConverter.convert(
+            data.disk_usage, UnitOfInformation.BYTES, UnitOfInformation.MEGABYTES
+        )
+    )
+    extra_state_attributes["disk_limit"] = round(
+        InformationConverter.convert(
+            data.disk_limit, UnitOfInformation.BYTES, UnitOfInformation.MEGABYTES
+        )
+    )
+
+    return extra_state_attributes
+
+
+SENSOR_DESCRIPTIONS = [
+    PterodactylSensorEntityDescription(
+        key=KEY_CPU_UTILIZATION,
+        translation_key=KEY_CPU_UTILIZATION,
+        value_fn=lambda data: data.cpu_utilization,
+        attributes_fn=get_extra_state_attributes_cpu,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        native_unit_of_measurement=PERCENTAGE,
+        suggested_display_precision=0,
+    ),
+    PterodactylSensorEntityDescription(
+        key=KEY_MEMORY_UTILIZATION,
+        translation_key=KEY_MEMORY_UTILIZATION,
+        value_fn=lambda data: ((data.memory_usage / data.memory_limit) * 100),
```

Comment:
I wouldn't add calculated values here imo

Reply:
But in api.py it would be fine, right?

Reply:
I removed the calculated entities.